zerver/decorator.py

```diff
@@ -172,7 +172,7 @@ def _wrapped_view_func(request, api_key=REQ,
                            *args, **kwargs):
 
         try:
-            user_profile = UserProfile.objects.get(api_key=api_key, is_active=True)
+            user_profile = UserProfile.objects.get(api_key=api_key, is_active=True, is_incoming_webhook=True)
         except UserProfile.DoesNotExist:
             raise JsonableError("Invalid API key")
```

Comment:
I think for backwards compatibility, we should not check for `is_incoming_webhook` here.

Instead, what I had in mind, is requiring `is_incoming_webhook=False` in these two functions:
* authenticate_log_and_execute_json
* validate_api_key (used in authenticated_rest_api_view and authenticated_api_view)
(thus preventing the `is_incoming_webhook bots from using the non-webhook parts of the API)